pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,503 @@
+					}
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": id,
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: type inference fail",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", Reader(""))
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "unable to parse response data" {
+					t.Errorf("Want: %v, Got: %v", "unable to parse response data", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: json failure",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+
+					json.NewEncoder(w).Encode("")
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "json: cannot unmarshal string into Go value of type model.Response" {
+					t.Errorf("Want: %v, Got: %v", "json: cannot unmarshal string into Go value of type model.Response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect id in response",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					_, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					err := r.ParseMultipartForm(10000)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file size exceeeded", nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file parse fail", nil)
+						log.Error(err.Error())
+						return
+					}
+					defer file.Close()
+
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": "gfhgv",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "incorrect id received in response" {
+					t.Errorf("Want: %v, Got: %v", "incorrect id received in response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect status code",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					err := r.ParseMultipartForm(10000) //File size to come from config
```

Comment:
If the parsed data is not required, remove the logic to parse the file data.
Looks like currently, it is reading the data using FormFile() and closing it without doing anything.

This being a failure case, only put the required logic. Not sure where the test server returns the status code as 404, which is being validated in the ValidateFunc.

Reply:
done